core-plugins/src/test/java/co/cask/hydrator/plugin/batch/action/OracleExportActionTest.java

```diff
@@ -0,0 +1,144 @@
+    OracleExportAction.OracleExportActionConfig oracleExportActionConfig =
+      new OracleExportAction.OracleExportActionConfig(sshServer.getHost(),
+                                                      sshServer.getPort(), USER, "wrongPassword", "system",
+                                                      "cask", "home", "sid", "/tmp/junit.csv",
+                                                      "select * from test;", "/tmp", "csv");
+    OracleExportAction oracleExportAction = new OracleExportAction(oracleExportActionConfig);
+    MockActionContext mockActionContext = new MockActionContext();
+    try {
+      oracleExportAction.run(mockActionContext);
+    } catch (IllegalStateException e) {
+      Assert.assertEquals(String.format("SSH authentication error when connecting to %s@%s on port %d",
+                                        USER, sshServer.getHost(),
+                                        sshServer.getPort()), e.getMessage());
+    }
+  }
+
+  @Test
+  public void testCheckWrongArgumentForFormat() throws Exception {
+    OracleExportAction.OracleExportActionConfig oracleExportActionConfig =
+      new OracleExportAction.OracleExportActionConfig(sshServer.getHost(),
+                                                      sshServer.getPort(), USER, PASSPHRASE, "system",
+                                                      "cask", "home", "sid", "/tmp/junit.csv",
+                                                      "select * from test;", "/tmp", "format");
+
+    try {
+    MockPipelineConfigurer configurer = new MockPipelineConfigurer(null);
+    new OracleExportAction(oracleExportActionConfig).configurePipeline(configurer);
+    } catch (IllegalArgumentException e) {
+      Assert.assertEquals(String.format("Invalid format '%s'. Must be one of %s", "format",
+                      EnumSet.allOf(OracleExportAction.SeparatorFormat.class)), e.getMessage());
+    }
+  }
+
+  @Test
+  public void testOracleExportAction() throws Exception {
+    Path outputDir = dfsCluster.getFileSystem().getHomeDirectory();
+    OracleExportAction.OracleExportActionConfig oracleExportActionConfig =
+      new OracleExportAction.OracleExportActionConfig(sshServer.getHost(),
+                                                      sshServer.getPort(), USER, PASSPHRASE, "system",
+                                                      "cask", "home", "sid",
+                                                      outputDir.toUri().toString() + "/tmp/OracleExportJunitTemp.csv",
+                                                      "select * from test;", "/tmp", "csv");
+    OracleExportAction oracleExportAction = new OracleExportAction(oracleExportActionConfig);
+    MockActionContext mockActionContext = new MockActionContext();
+    oracleExportAction.run(mockActionContext);
+    Assert.assertTrue(fileSystem.exists(new Path(outputDir.toUri().toString() + "/tmp/OracleExportJunitTemp.csv")));
```

Comment:
there's no oracle here, right? So in that case, this will just generate an empty csv file?